File: Day9/QuickSort.java
package Day9;

public class QuickSort {
    public static void main(String[] args) {
        int arr[]={51,95,66,72,42,38,39,41,15};
        int left=0;
        int right=arr.length-1;
        
        quicksort(arr,left,right);

    }
    public static void quicksort(int arr[],int left,int right)
    {
       if(left<right)
       {
          int p=partition(arr,left,right);
          quicksort(arr,left,p-1);
          quicksort(arr,p+1,right);
       }
       for(int i=0;i<arr.length;i++)
       {
          System.out.print(arr[i]+" ");
       }
       System.out.println("");
    }
    public static int partition(int arr[],int left,int right)
    {
        int pivot=arr[left];
        int count=0;
        for(int i=left+1;i<=right;i++)
        {
            if(arr[i]<pivot)
            {
                count++;
            }
        }
        int p=left+count;
        int temp=arr[left];
        arr[left]=arr[p];
        arr[p]=temp;
        int i=left,j=right;
        while(i<p&&j>p)
        {
            while(arr[i]<pivot)
                i++;
            while(arr[j]>pivot)
                j--;
            int temp1=arr[i];
            arr[i]=arr[j];
            arr[j]=temp1;
        }
        return p;
    }

}

File: Day9/QuickSort.py

def QuickSort(arr,left,right):
    if(left<right):
        p=partition(arr,left,right)
        QuickSort(arr,left,p-1)
        QuickSort(arr,p+1,right)

def partition(arr,left,rigth):
    pivot=arr[left]
    count=0
    for i in range(left+1,right+1):
        if(arr[i]<pivot):
            count=count+1
    p=left+count
    temp=arr[left]
    arr[left]=arr[p]
    arr[p]=temp
    i=left
    j=right
    while (i<p and j>p):
        while(arr[i]<pivot):
            i=i+1
        while(arr[j]>pivot):
            j=j-1
        temp1=arr[i]
        arr[i]=arr[j]
        arr[j]=temp1
    return p


arr=[51,95,66,72,42,38,39,41,15]
left=0
right=len(arr)-1
QuickSort(arr,left,right)
for i in range(0,len(arr)):
    print(arr[i],end=" ")


